controllers/spacerequest/spacerequest_controller.go

```diff
@@ -160,6 +160,33 @@ func (r *Reconciler) addFinalizer(ctx context.Context, memberCluster cluster.Clu
 	return nil
 }
 
+// fetches the subspace corresponding to the space request given.
+//
+// returns:
+//   - nil, err if an error occurred retrieving spaces or more than one matching subspace was found
+//   - nil, nil if no subspaces exist
+//   - space, nil if a matching space was found
+func (r *Reconciler) getSubSpace(ctx context.Context, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, error) {
```

Comment:
minor thing feel free to leave it as is - but how about `listSubSpace`? Since it's actually listing and not getting.